### PR TITLE
HCA Form Config Chapter/Page Refactor

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
Refactor HCA form config from being in one file to multiple files separated by chapter then page.

[Ticket](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/17071)

## Testing done
- Manual, visual, and automated regression testing

http://be3e9dca95351a7bca8495287c8166ae.review.vetsgov-internal/health-care/apply/application/introduction

## Screenshots
N/A

## Acceptance criteria
- [x] Break up HCA form into more maintainable files

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/hca/config/chapters/householdInformation/annualIncome.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            (formData.maritalStatus.toLowerCase() === 'married' ||
+              formData.maritalStatus.toLowerCase() === 'separated'),
+          'ui:validations': [validateCurrency],
+        },
+      ),
+      spouseOtherIncome: merge(currencyUI('Spouse other income amount'), {
+        'ui:required': formData =>
+          formData.maritalStatus &&
+          (formData.maritalStatus.toLowerCase() === 'married' ||
+            formData.maritalStatus.toLowerCase() === 'separated'),
```
Comment (Mottie) [inline src/applications/hca/config/chapters/householdInformation/annualIncome.js:83]: It'd be nice if these 4 repeated checks were made into a function

File: src/applications/hca/config/chapters/householdInformation/dependentInformation.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    dependents: {
+      items: dependentUI,
+      'ui:options': {
+        expandUnder: 'view:reportDependents',
+        itemName: 'Dependent',
+        hideTitle: true,
+        viewField: DependentView,
+      },
+      'ui:errorMessages': {
+        minItems: 'You must add at least one dependent.',
```
Comment (Mottie) [inline src/applications/hca/config/chapters/householdInformation/dependentInformation.js:27]: I couldn't find an exact reference, but "must" is too strong of a language to use for feedback... the other feedback messaging examples](https://design.va.gov/patterns/messaging-dictionary) start with "please"... I'd get feedback on this from your content specialist.

File: src/applications/hca/config/chapters/militaryService/additionalInformation.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    radiumTreatments: {
+      'ui:title':
+        'Received nose/throat radium treatments while in the military',
+    },
+    campLejeune: {
+      'ui:title':
+        'Served on active duty at least 30 days at Camp Lejeune from January 1, 1953, through December 31, 1987',
+    },
+  },
```
Comment (Mottie) [inline src/applications/hca/config/chapters/militaryService/additionalInformation.js:57]: Does this page do anything? All I see are titles.

File: src/applications/hca/config/chapters/veteranInformation/personalnformation.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            'Please provide a valid name. Must be at least 1 character.',
+          pattern: 'Please provide a valid name. Must be at least 1 character.',
+        },
+      },
+      last: {
+        'ui:errorMessages': {
+          minLength:
+            'Please provide a valid name. Must be at least 2 characters.',
+          pattern:
+            'Please provide a valid name. Must be at least 2 characters.',
```
Comment (Mottie) [inline src/applications/hca/config/chapters/veteranInformation/personalnformation.js:28]: Lots of "must"s here too.

**Review by Mottie — APPROVED** [review]
ghe-emoji

File: src/applications/hca/config/chapters/militaryService/additionalInformation.js (hunk 3 above)
Comment (LinnJS) [reply src/applications/hca/config/chapters/militaryService/additionalInformation.js:55]: Good question that code did look pretty odd and on GH I didn't see where the `vets-json-schema` was being pulled in, but it looks like just a bunch of boolean values... Could probably be refactored into a `checkbox group` but what are the pros & cons of a `checkbox group`? Would it add value?

image

File: src/applications/hca/config/chapters/householdInformation/dependentInformation.js (hunk 2 above)
Comment (LinnJS) [reply src/applications/hca/config/chapters/householdInformation/dependentInformation.js:24]: Thanks for pointing this out I know @jonathanjnelson is planning on doing a overhaul of the content on this form... Will get with him and open another `tech debt` ticket to overview all error handling messages across the form.

File: src/applications/hca/config/chapters/householdInformation/annualIncome.js (hunk 1 above)
Comment (LinnJS) [reply src/applications/hca/config/chapters/householdInformation/annualIncome.js:79]: @Mottie I broke this logic into funcs to be more readable. If you dont mind double checking my logic, you know how booleans can be sometimes.

File: src/applications/hca/config/chapters/militaryService/additionalInformation.js (hunk 3 above)
Comment (Mottie) [reply src/applications/hca/config/chapters/militaryService/additionalInformation.js:55]: When the checkbox group is used, it wraps all the checkboxes in a single `<fieldset>`, I think how it is now, each checkbox is wrapped in it's own `<fieldset>` which isn't ideal.

Comment (LinnJS) [reply src/applications/hca/config/chapters/militaryService/additionalInformation.js:55]: Roger that will create another issue for this refactor

File: src/applications/hca/config/chapters/householdInformation/annualIncome.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      'ui:validations',
+      [validateCurrency],
+      currencyUI('Veteran\u2019s other income amount'),
+    ),
+    'view:spouseIncome': {
+      'ui:title': 'Spouse\u2019s income',
+      'ui:options': {
+        hideIf: formData =>
+          !formData.maritalStatus ||
+          (!isVeteranMarried(formData) && !isVeteranSeparated(formData)),
```
Comment (Mottie) [inline src/applications/hca/config/chapters/householdInformation/annualIncome.js:60]: This took me a minute, but now it makes sense 😅

File: src/applications/hca/config/chapters/insuranceInformation/general.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      'ui:widget': 'yesNo',
+    },
+    providers: {
+      'ui:options': {
+        itemName: 'Insurance Policy',
+        expandUnder: 'isCoveredByHealthInsurance',
+        viewField: InsuranceProviderView,
+      },
+      'ui:errorMessages': {
+        minItems: 'You need to at least one provider.',
```
Comment (Mottie) [inline src/applications/hca/config/chapters/insuranceInformation/general.js:24]: Language is weird here... Use `Please add at least one provider?` maybe?

**Review by Mottie — APPROVED** [review]

File: src/applications/hca/config/chapters/householdInformation/annualIncome.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+const isVeteranMarried = formData =>
+  formData.maritalStatus?.toLowerCase() === 'married';
+
+const isVeteranSeparated = formData =>
+  formData.maritalStatus?.toLowerCase() === 'separated';
+
+const hasVeteranBeenMarried = formData =>
+  (formData.maritalStatus && isVeteranMarried(formData)) ||
+  isVeteranSeparated(formData);
```
[the remark this answers is not in the document]
Comment (LinnJS) [reply src/applications/hca/config/chapters/householdInformation/annualIncome.js:22]: Good catch

File: src/applications/hca/config/chapters/householdInformation/annualIncome.js (hunk 5 above)
Comment (LinnJS) [reply src/applications/hca/config/chapters/householdInformation/annualIncome.js:60]: @Mottie do you have any suggestions on making it easier to digest?

File: src/applications/hca/config/chapters/insuranceInformation/general.js (hunk 6 above)
Comment (LinnJS) [reply src/applications/hca/config/chapters/insuranceInformation/general.js:24]: Yess sir made a [ticket](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/19435) to go over all content particularly validation message here as per your feedback

Also made this [ticket](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/19438) for checkbox groups, @Mottie

File: src/applications/hca/config/chapters/householdInformation/annualIncome.js (hunk 5 above)
Comment (Mottie) [reply src/applications/hca/config/chapters/householdInformation/annualIncome.js:60]: Naw, the `hideIf` function always throws me off... it's fine as it is

Comment (LinnJS) [reply src/applications/hca/config/chapters/householdInformation/annualIncome.js:60]: Yeah I feel that my brain hurts every time I switch between `depends` and `hideIf`

File: src/applications/hca/config/form.js [hunk 8]
```diff
@@ -250,77 +125,15 @@ const formConfig = {
           path: 'veteran-information/personal-information',
           title: 'Veteran information',
           initialData: {},
```
Comment (craigwheeler) [inline src/applications/hca/config/form.js:127]: are we using the `initialData: {}` prop or can this be removed?

Comment (LinnJS) [reply src/applications/hca/config/form.js:127]: Looks like if I remove those it triggers contract errors for 508 in CY Axe Check

**Review by jonathanjnelson — APPROVED** [review]
Looks good!

**Pull request merged**